src/main/java/com/iotiq/application/service/SellerService.java

```diff
@@ -10,19 +12,31 @@
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
 @Service
 @RequiredArgsConstructor
 public class SellerService {
 
     private final SellerRepository sellerRepository;
     private final UserService userService;
+    private final OrderedProductService orderedProductService;
 
     @Transactional
     public Seller getCurrentSellerOrCreate() {
         User currentUser = userService.getCurrentUser();
         return sellerRepository.findByUser(currentUser).orElseGet(() -> createDefaultSeller(currentUser));
     }
 
+    @Transactional
```

Comment:
Does this need to be transactional?

Reply:
We used getCurrentSellerOrCreate inside the method

Reply:
Why are we creating something in the get method?

Reply:
I opened a new issue for this https://github.com/Iotiq-Erste/kimono-application/issues/48